hooks/tk-multi-publish2/maya/publish_files_integ.py

```diff
@@ -0,0 +1,266 @@
+                used to check node hierarchy once camera naming is validated.
+        """
+        # Look for all the cameras present in the scene
+        all_cameras = cmds.listCameras()
+        # Remove the default_cameras from the list
+        main_cam = list(set(all_cameras) - set(_default_cameras))
+        if main_cam:
+            if len(main_cam) > 1:
+                # Checking if more than one CAM present
+                self.logger.error("More the one camera detected. Only CAM should be present.")
+                return False
+            elif main_cam[0] != _camera_name:
+                # Validating camera name
+                self.logger.error("Incorrect camera name! Should be CAM.")
+                return False
+        else:
+            self.logger.error("Camera (CAM) not present in the scene.")
+            return False
+        return self._connected_image_plane(main_cam[0])
+
+    def _node_naming(self, group_nodes, groups):
+        """Checking if the established group node names have not been tampered with.
+            :param:
+                group_nodes: group nodes to be present in the scene
+                groups: group nodes that are actually present
+        """
+        # Check for extra group nodes apart from the ones in group_nodes
+        names = ""
+        extras = list(set(groups) - set(group_nodes))
+        # check for any group nodes apart from the once mentioned
+        if extras:
+            for item in extras:
+                names += "\n" + item
+            self.logger.error("Incorrect naming or extra group nodes present in the scene.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Show the conflicting group nodes",
+                                      "text": "Please check the group nodes:\n{}".format(names) +
+                                              "\n\nOnly following group nodes should be present:\nWORLDSCALE \nSET_TO_WORLD \nTRACK_GEO"
+                                  }
+                              }
+                              )
+            return False
+        # check if any of the required group nodes are missing
+        elif not set(group_nodes).issubset(set(groups)):
+            self.logger.error("Please ensure all the group nodes are present.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Group nodes",
+                                      "text": "Following group nodes should be present:\nWORLDSCALE \nSET_TO_WORLD \nTRACK_GEO"
+                                  }
+                              }
+                              )
+            return False
+        return self._check_hierarchy(group_nodes)
```

Comment:
Shouldn't be getting called from the top-level validation() method?

Reply:
Also, shouldn't you be passing the actual scene group nodes and not the global constant list of possible group nodes?